pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -100,6 +112,73 @@ func (r *MachineSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // Reconcile reconciles CAPI and MAPI machines for their respective namespaces.
 func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx, "namespace", req.Namespace, "name", req.Name)
+
+	logger.V(1).Info("Reconciling machine")
+	defer logger.V(1).Info("Finished reconciling machine")
+
+	var (
+		mapiMachineNotFound bool
+		capiMachineNotFound bool
+	)
+
+	// Get the MAPI Machine
+	mapiMachine := &machinev1beta1.Machine{}
+	if err := r.Get(ctx, req.NamespacedName, mapiMachine); apierrors.IsNotFound(err) {
+		logger.Info("MAPI Machine not found")
+
+		mapiMachineNotFound = true
+	} else if err != nil {
+		logger.Error(err, "Failed to get MAPI Machine")
+		return ctrl.Result{}, fmt.Errorf("%w: %w", errFailedToGetMAPIMachine, err)
```

Comment:
Last time I did this double wrapping, @JoelSpeed wasn't too fond of it, is that still the case Joel?

Reply:
Yeah, I think for now lets keep to writing them out and only wrapping the base error. Until and unless we find some good reason we need to have a static error here
```suggestion
		return ctrl.Result{}, fmt.Errorf("failed to get MAPI machine: %w", err)
```

Reply:
What about in cases where I don't have an error to return, then I tend to get linter errors :(  e.g: 

``` 
do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unexpected value for spec.AuthoritativeAPI: %s\", mapiMachine.Spec.AuthoritativeAPI)" (err113)
                return ctrl.Result{}, fmt.Errorf("unexpected value for spec.AuthoritativeAPI: %s", mapiMachine.Spec.AuthoritativeAPI)
```

Reply:
Yes in that case, create a static error, but when you're just wrapping an existing error there is no need to create additional ones unless you plan to unwrap them later. The idea is so that we have a base error that can be unwrapped to find the root cause